### ISSUE TITLE
Deleting a volume and adding a new volume with the same name, and creating a service to use the new volume errors out

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#8607]
**Rancher Versions:**
Server: Master

**Docker Version:**1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**Cloud

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

**Steps to Reproduce:**
1. Deploy rancher-ebs catalog
2. Add a volume volume1 through UI
3. Create a service to use the volume1 and ensure you can access the volume for read and write
4. Delete the service and the volume through UI
5. Create another volume with name volume1 and create a service to use the volume

**Results:** 
Service creation fails with error 
```Expected state running but got error: /VolumeDriver.Attach 'volume1' (driver 'rancher-ebs') returned status 500: 500 Internal Server Error```

ebs-driver logs below:

```
4/27/2017 11:06:28 AMtime="2017-04-27T18:06:28Z" level=error msg=attach.response error="No such volume"
4/27/2017 11:06:28 AMtime="2017-04-27T18:06:28Z" level=info msg=attach.request name=volume1
4/27/2017 11:06:28 AMtime="2017-04-27T18:06:28Z" level=error msg=attach.response error="No such volume"
4/27/2017 11:06:29 AMtime="2017-04-27T18:06:29Z" level=info msg=attach.request name=volume1
4/27/2017 11:06:29 AMtime="2017-04-27T18:06:29Z" level=error msg=attach.response error="No such volume"
4/27/2017 11:06:29 AMtime="2017-04-27T18:06:29Z" level=info msg=attach.request name=volume1
4/27/2017 11:06:29 AMtime="2017-04-27T18:06:29Z" level=error msg=attach.response error="No such volume"


```

**Comment by StrongMonkey** [comment]
It is available at rancher/storage-ebs:v0.8.1.

**Comment by soumyalj** [comment]
Tested with rancher 1.6.0-rc3 using rancher/storage-ebs:v0.8.1. 
1. Deployed rancher-ebs catalog and added a volume volume1 through UI
2. Created a service to use the volume1 and verified read/write access
3. Deleted the service and the volume through UI
4. Created another volume with name volume1 and a service to use the volume. 
The service creation was successful. Issue above was not found.
